src/viam/sdk/components/motor/client.cpp

```diff
@@ -141,66 +81,27 @@ grpc::StatusCode MotorClient::stop(const AttributeMap& extra) {
 }
 
 Motor::power_status MotorClient::get_power_status(const AttributeMap& extra) {
-    viam::component::motor::v1::IsPoweredRequest request;
-    viam::component::motor::v1::IsPoweredResponse response;
-
-    grpc::ClientContext ctx;
-    set_client_ctx_authority(ctx);
-
-    *request.mutable_name() = this->name();
-    *request.mutable_extra() = map_to_struct(extra);
-
-    const grpc::Status status = stub_->IsPowered(&ctx, request, &response);
-    if (!status.ok()) {
-        throw std::runtime_error(status.error_message());
-    }
-    return from_proto(response);
+    return make_client_helper(this, *stub_, &StubType::IsPowered)
```

Comment:
⬆️  - I did not convert `MotorClient::stop` because the behavior is non-standard. As written the `ClientHelper` cannot deal with it, because it splits the status into the good case where you handle a response and a bad case where you handle an error. It may actually be possible to SFINAE in the ability to have the invoke handler take a callback that takes a `grpc::Status` object, which would let us use the helper here.

Reply:
Yeah I would rather change the `stop` return type than SFINAE our way around it if possible.